scale/recipe/views.py

```diff
@@ -631,9 +632,16 @@ def _post_v6(self, request, recipe_id):
             raise BadParameter(unicode(ex))
 
         try:
-            recipe = Recipe.objects.select_related('recipe_type', 'recipe_type_rev').get(id=recipe_id)
+            recipe = Recipe.objects.select_related('recipe_type').get(id=recipe_id)
+            if revision_num:
+                recipe.recipe_type_rev = RecipeTypeRevision.objects.get_revision(recipe.recipe_type.name, revision_num)                
+            else:
+                revision_num = recipe.recipe_type.revision_num
+                recipe.recipe_type_rev = RecipeTypeRevision.objects.get_revision(recipe.recipe_type.name, recipe.recipe_type.revision_num)
```

Comment:
Add comment to clarify that we're checking if the revision exists and use a throwaway variable to store the result rather than making it look like we're updating the recipe we got with a new revision object. Also, I don't think we need to check if the revision number exists if we're using the existing recipe type's revision_num variable.